kiliautoml/models/_pytorchvision_image_classification.py

```diff
@@ -51,117 +57,190 @@ def __init__(
 
         model_dir = PathPytorchVision.append_model_dir(model_repository_dir)
         model_path = PathPytorchVision.append_model_path(model_repository_dir, model_name)
-        data_dir = os.path.join(model_repository_dir, "data")
-        download_project_image_clean_lab(
-            assets=assets,
-            api_key=api_key,
-            data_path=data_dir,
-            job_name=job_name,
-        )
+        data_dir = PathPytorchVision.append_data_dir(model_repository_dir)
 
         # To set to False if the input size varies a lot and you see that the training takes
         # too much time
         cudnn.benchmark = True
 
-        original_image_datasets = get_original_image_dataset(data_dir)
-
-        class_names = original_image_datasets["train"].classes  # type: ignore
-        labels = [label for img, label in datasets.ImageFolder(data_dir).imgs]
-        assert len(class_names) > 1, "There should be at least 2 classes in the dataset."
+        BaseModel.__init__(
+            self,
+            job=job,
+            job_name=job_name,
+            model_name=model_name,
+            model_framework=model_framework,
+        )
 
-        self.model_name: ModelNameT = model_name
         self.model_dir = model_dir
         self.model_path = model_path
         self.data_dir = data_dir
-        self.original_image_datasets = original_image_datasets
-        self.class_names = class_names
-        self.labels = labels
-
-    def train(self, epochs, verbose):
 
-        image_datasets = copy.deepcopy(self.original_image_datasets)
-        train_idx, val_idx = train_test_split(
-            range(len(self.labels)), test_size=0.2, random_state=42
+    def prepare_dataset(
+        self,
+        assets: List[AssetT],
+        api_key,
+    ):
+        shutil.rmtree(self.data_dir, ignore_errors=True)
+        download_project_image_clean_lab(
+            assets=assets,
+            api_key=api_key,
+            data_path=self.data_dir,
+            job_name=self.job_name,
         )
+        original_image_datasets = get_original_image_dataset(self.data_dir)
+
+        class_names = original_image_datasets["train"].classes  # type: ignore
+        labels = [label for _, label in datasets.ImageFolder(self.data_dir).imgs]
+        assert len(class_names) > 1, "There should be at least 2 classes in the dataset."
+        return original_image_datasets, labels, class_names
+
+    def train(
+        self,
+        *,
+        assets: List[AssetT],
+        epochs: int,
+        batch_size: int,
+        clear_dataset_cache: bool,
+        disable_wandb: bool,
+        verbose: int = 1,
+        api_key: str = "",
+    ):
+        _ = clear_dataset_cache
+
+        if disable_wandb is False:
```

Comment:
Problem: `disable_wandb` is false by default, so the default parameters will make this kind of models crash. I suggest we replace the error by a warning here (unless you have a better option).